rts/Lua/LuaUnsyncedCtrl.cpp

```diff
@@ -3729,8 +3729,8 @@ int LuaUnsyncedCtrl::ShareResources(lua_State* L)
 		return 0;
 	}
 
-	if (args < 3)
-		return 0;
+	if ((args < 3) || !lua_isnumber(L, 3))
```

Comment:
I think `!lua_isnumber` makes the `args < 3` check redundant (the type of the 3rd value would be "none" in that case) but that's no biggie.
```suggestion
	if (!lua_isnumber(L, 3))
```

Reply:
applied your suggestion